libcaf_net/test/ip.cpp

```diff
@@ -30,16 +30,48 @@
 using namespace caf;
 using namespace caf::net;
 
-CAF_TEST_FIXTURE_SCOPE(ip_tests, host_fixture)
+namespace {
+
+struct fixture : host_fixture {
+  fixture() : v6_local{{0}, {0x1}} {
+    v4_local = ip_address{make_ipv4_address(127, 0, 0, 1)};
+    v4_any_addr = ip_address{make_ipv4_address(0, 0, 0, 0)};
+  }
+
+  bool contains(ip_address x) {
+    return std::count(addrs.begin(), addrs.end(), x) > 0;
+  }
+
+  ip_address v4_any_addr;
+  ip_address v6_any_addr;
+  ip_address v4_local;
+  ip_address v6_local;
+
+  std::vector<ip_address> addrs;
+};
+
+} // namespace
+
+CAF_TEST_FIXTURE_SCOPE(ip_tests, fixture)
 
 CAF_TEST(resolve) {
-  ip_address v4_local{make_ipv4_address(127, 0, 0, 1)};
-  ip_address v6_local{{0}, {0x1}};
-  auto addrs = ip::resolve("localhost");
+  addrs = ip::resolve("localhost");
+  CAF_CHECK(!addrs.empty());
+  CAF_CHECK(contains(v4_local) || contains(v6_local));
+}
+
+CAF_TEST(local addresses) {
+  CAF_MESSAGE("check: v4");
```

Comment:
Why not separate each of these checks into an own TEST instead? That would make it more readable IMO.

Reply:
Hm, I think the problem here is actually that the `local_address` call is wrong in returning an address when `::` is passed and the system doesn't have IPv6. Might be better to delegate calls with "localhost" or one of the "any" addresses to `resolve` which could handle this with #23.